src/flb_downstream.c

```diff
@@ -188,7 +188,7 @@ static int prepare_destroy_conn(struct flb_connection *connection)
     flb_trace("[downstream] destroy connection #%i to %s",
               connection->fd, flb_connection_get_remote_address(connection));
 
-    if (stream->flags & FLB_IO_ASYNC) {
+    if(MK_EVENT_IS_REGISTERED((&connection->event))) {
```

Comment:
I think the convention is to put a space after the `if`. Also, the need for the extra parentheses seems to be a bug in `mk_event.h`…
```suggestion
    if (MK_EVENT_IS_REGISTERED((&connection->event))) {
```